app/controllers/user_controller.php

```diff
@@ -0,0 +1,100 @@
+<?php
+class UserController extends AppController
+{
+    public function register()
+    {
+        $page = Param::get('page_next', 'register');
+        $error = false;
+
+        switch ($page) {
+            case 'register':
+                break;
+            case 'write_end':
+                $params = array(
+                    'firstname' => Param::get('firstname'),
+                    'lastname' => Param::get('lastname'),
+                    'email' => Param::get('email'),
+                    'username' => Param::get('username'),
+                    'password' => Param::get('password'),
+                    'validate_password' => Param::get('validate_password')
+                );
+                $user = new User($params);
+
+                if (!$this->isMatchPassword()) {
+                    $user->validation_errors['password']['match'] = true;
+                }
+
+                try {
+                    $user->register();  
+                } catch (ValidationException $e) {
+                    $page = 'register';
+                }    
+                break;
+            default:
+                throw new NotFoundException("{$page} is not found");
+                break;
+        }
+        $this->set(get_defined_vars());
+        $this->render($page);
+    }
+
+    public function login()
+    {
+        $login_info = new LoginInfo;
+        $user = new User;
+        $page = Param::get('page_next', 'login');
+        $error = false;
+
+        if (isset($_SESSION['username'])) {
+            redirect('/user/profile');
+        }
+
+        switch ($page) {
+            case 'login':
+                break;
+            case 'write_end':
+                $login_info->username = Param::get('username');
+                $login_info->password = Param::get('password');
+
+                $user->username = Param::get('username');
+                $user->password = Param::get('password');
+                
```

Comment:
For L56-57, and L59-60, please apply this [comment] (https://github.com/ezrafinones/board/pull/1#discussion_r27191024)